perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/BBMRI.java

```diff
@@ -0,0 +1,161 @@
+
+	final static Logger log = LoggerFactory.getLogger(BBMRI.class);
+	public static final String BIOBANK_IDS_FIELD = "Comma or new-line separated list of collection IDs you are representing:";
+
+	private RegistrarManager registrar;
+
+	@Override
+	public void setRegistrar(RegistrarManager registrar) {
+		this.registrar = registrar;
+	}
+
+	@Override
+	public List<ApplicationFormItemData> createApplication(PerunSession user, Application application, List<ApplicationFormItemData> data) throws PerunException {
+		return data;
+	}
+
+	/**
+	 * Add users to the listed groups.
+	 */
+	@Override
+	public Application approveApplication(PerunSession session, Application app) throws PerunException {
+		// get perun and beans from session
+		Perun perun = session.getPerun();
+		Vo vo = app.getVo();
+		User user = app.getUser();
+		Member member = perun.getMembersManager().getMemberByUser(session, vo, user);
+
+		// get the field of application with the collections
+		Set<String> collectionIDsInApplication = getCollectionIDsFromApplication(session, app);
+
+		// get map of collection IDs to group from Perun
+		Map<String, Group> collectionIDsToGroupsMap = new HashMap<>();
+		Group collectionsGroup = perun.getGroupsManager().getGroupByName(session, vo, "collections");
+		for (Group group : perun.getGroupsManager().getSubGroups(session, collectionsGroup)) {
+			for (Group subgroup : perun.getGroupsManager().getSubGroups(session, group)) {
+				if ("representatives".equals(subgroup.getShortName())) {
+					Attribute collectionID = perun.getAttributesManager().getAttribute(session, subgroup, "urn:perun:group:attribute-def:def:collectionID");
+					collectionIDsToGroupsMap.put((String) collectionID.getValue(), subgroup);
+				}
+			}
+		}
+
+		// add user to all groups from the field on application
+		for (String collectionID : collectionIDsInApplication) {
+			perun.getGroupsManager().addMember(session, collectionIDsToGroupsMap.get(collectionID), member);
+		}
+
+		return app;
+	}
+
+	@Override
+	public Application rejectApplication(PerunSession session, Application app, String reason) throws PerunException {
+		return app;
+	}
+
+
+	/**
+	 * Checks whether all collection IDs found in user input really exists in Perun.
+	 * If not, CantBeApproved exception is thrown.
+	 *
+	 * @param session who approves the application
+	 * @param app     application
+	 * @return application
+	 * @throws CantBeApprovedException if at least one collection ID does not exist in Perun
+	 */
+	public Application beforeApprove(PerunSession session, Application app) throws PerunException {
+		// get perun and beans from session
+		Perun perun = session.getPerun();
+		Vo vo = app.getVo();
+
+		// get all collection IDs from Perun
+		Set<String> collectionIDsInPerun = new HashSet<>();
+		Group collectionsGroup = perun.getGroupsManager().getGroupByName(session, vo, "collections");
+		for (Group group : perun.getGroupsManager().getSubGroups(session, collectionsGroup)) {
```

Comment:
Just for consideration, you can also use: 
```java
List<RichGroup> groups = perun.getGroupsManager().getAllRichSubGroupsWithAttributesByNames(session, collectionsGroup, Arrays.asList("urn:perun:group:attribute-def:def:collectionID"));
```
which will return whole sub-group hierarchy sorted. It probably depends whether you expect group structure to change in a future or "representatives" groups should be always on a third level. If group structure will be fixed, your solution might be even quicker.